compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -659,7 +659,7 @@ fn infer_placeholder_type(
                 format!("{}: {}", item_ident, ty),
                 Applicability::MachineApplicable,
             )
-            .emit();
+            .emit_unless(matches!(ty.kind(), ty::Error(_)));
```

Comment:
I think we might want to check if `ty.references_error()` here and below in the `None` case instead. That way we account for things like `Option<[type error]>` as well.